File: sol/2109/2109.py
class Solution:
    def addSpaces(self, s: str, spaces: List[int]) -> str:
        prevIdx = 0
        buff = []
        for idx in spaces:
            buff.append(s[prevIdx:idx])
            prevIdx = idx
        buff.append(s[prevIdx:])
        return " ".join(buff)

File: sol/2109/2109.cpp
struct Solution {
    string addSpaces(string &s, vector<int>& spaces) {
        int n = s.size(), ptr = 0, l = spaces.size(), wptr = 0;
        string res(n + l, 0);
        for (int i = 0; i < n; ++i){
            if (ptr < l && i == spaces[ptr]){
                res[wptr++] = ' ';
                ++ptr;
            }
            res[wptr++] = s[i];
        }
        return res;
    }
};
